react/src/components/BAICodeEditor.tsx

```diff
@@ -1,35 +1,36 @@
+import useControllableState from '../hooks/useControllableState';
 import { loadLanguage, LanguageName } from '@uiw/codemirror-extensions-langs';
 import CodeMirror, { ReactCodeMirrorProps } from '@uiw/react-codemirror';
 import { EditorView } from '@uiw/react-codemirror';
-import { useControllableValue } from 'ahooks';
 
 interface BAICodeEditorProps extends Omit<ReactCodeMirrorProps, 'language'> {
-  children: string;
   onChange: (value: string) => void;
+  value: string;
   language: LanguageName;
   editable?: boolean;
   showLineNumbers?: boolean;
   lineWrapping?: boolean;
 }
 
 const BAICodeEditor: React.FC<BAICodeEditorProps> = ({
-  children,
   onChange,
+  value,
   language = 'shell',
   editable = false,
   showLineNumbers = true,
   lineWrapping = false,
   ...CodeMirrorProps
 }) => {
-  const [script, setScript] = useControllableValue<string>({
+  const [script, setScript] = useControllableState<string>({
     defaultValue: '',
-    value: children,
+    value,
     onChange,
   });
   const extensions = [loadLanguage(language)!];
 
   return (
     <CodeMirror
+      {...CodeMirrorProps}
```

Comment:
The properties below, such as `basicSetup`, `theme`, etc., cannot be overridden. Move the items that should not be overridden below `{...CodeMirrorProps}`, and move the items that can be overridden above `{...CodeMirrorProps}`.

Reply:
I modified it ! https://github.com/lablup/backend.ai-webui/pull/2572/files#diff-a6f793908a50d0ec4d6fec1664904acc0e6eb3ba89bb374f697c3ea3ff739ddeR42